BrushUpCode/src/Coffee.java

```diff
@@ -0,0 +1,272 @@
+        System.out.println("Write how many disposable cups of coffee do you want to add: ");
+        int addedCups = enter.nextInt();
+        cups += addedCups;
+        setCups(cups);
+
+    }
+
+    /*method for instruction buy, consists of three different methods to buy espresso latte and cappuccino*/
+    private void buy(int dollar, int water, int milk, int coffeeBeans, int cups) {
+        Scanner enter = new Scanner(System.in);
+        System.out.println("What do you want to buy? 1) espresso 2) latte 3) cappuccino ? If mistaken you can go back by instruction  4) back");
+        int choice = enter.nextInt();
+        this.coffeeMachineState = State.MAKINGCOFFE;
+        switch (choice) {
+            case 1: {
+                System.out.println("Good Choice! ");
+                buyEspresso(dollar, water, milk, coffeeBeans, cups);
+                break;
+            }
+
+            case 2: {
+                System.out.println("Yummy Latte!");
+
+                buyLatte(dollar, water, milk, coffeeBeans, cups);
+                break;
+            }
+
+            case 3: {
+                System.out.println("Great choice!");
+                buyCappuccino(dollar, water, milk, coffeeBeans, cups);
+                break;
+            }
+            case 4: {
+                break;
+            }
+        }
+    }
+
+
+    /*method for buying espresso(used in method buy)*/
+    private void buyEspresso(int dollar, int water, int milk, int coffeeBeans, int cups) {
+        enoughForCoffee("espresso");
+        water -= espressoWater;
+        setWater(water);
+        coffeeBeans -= espressoCoffee;
+        setCoffeeBeans(coffeeBeans);
+        dollar += espressoMoney;
+        setDollar(dollar);
+        cups -= 1;
+        setCups(cups);
+    }
+
+    /*method for buying Latte(used in method buy)*/
+    private void buyLatte(int dollar, int water, int milk, int coffeeBeans, int cups) {
+        enoughForCoffee("latte");
+        water -= latteWater;
+        setWater(water);
+        milk -= latteMilk;
+        setMilk(milk);
+        coffeeBeans -= latteCoffee;
+        setCoffeeBeans(coffeeBeans);
+        dollar += latteMoney;
+        setDollar(dollar);
+        cups -= 1;
+        setCups(cups);
+
+    }
+
+    /*method for buying Cappuccino(used in method buy)*/
+    private void buyCappuccino(int dollar, int water, int milk, int coffeeBeans, int cups) {
+        enoughForCoffee("cappuccino");
+        water -= cappuccinoWater;
+        setWater(water);
+        milk -= cappuccinoMilk;
+        setMilk(milk);
+        coffeeBeans -= cappuccinoCoffee;
+        setCoffeeBeans(coffeeBeans);
+        dollar += cappuccinoMoney;
+        setDollar(dollar);
+        cups -= 1;
+        setCups(cups);
+
+    }
+
+
+    private void enoughForCoffee(String coffeeType) {
+        if (coffeeType.equals("espresso")) {
+            if (water >= espressoWater && coffeeBeans >= espressoCoffee && cups >= 1)
```

Comment:
@maria-salmasian jan inner if and inner if in else can be seperate methods I guess , method is too big

Reply:
I did not have this method at first, then Hayk said it would be better to have a method for it. I dont know how to make it smaller, I have different variables for latteWater EspressoWater etc. please let me know what to do

Reply:
@maria-salmasian for example you can create one method for espresso type and if it is espresso call that method etc. the same for others

Reply:
broke down the method enough for coffee into 3 methods: enoughForLatte, enoughForEspresso, enoughForCappuccino